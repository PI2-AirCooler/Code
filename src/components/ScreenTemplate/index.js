
import React from 'react';
import {
  Container,
  TitleText,
  HeaderContainer,
  BodyContainer,
  FooterContainer,
  LeftButton,
  RightButton,
  BlankView,
  TitleTextView,
} from './style';
import Button from '../Button';
import Icon from 'react-native-vector-icons/MaterialIcons';
Icon.loadFont();

function ScreenTemplate({
  MainText,
  children,
  headerColor,
  footerColor,
  buttonColor,
  buttonText,
  onPress,
  hasButton = true,
  onPressRightButton,
  hasLeftButton = false,
  hasRightButton = false,
  blankViewColor,
  onBack,
  MainTextColor,
  RightButtonBackgroundColor,
  RightButtonIconColor,
}) {
  return (
    <Container>
      <HeaderContainer color={headerColor}>
        {hasLeftButton ? (
          <LeftButton onPress={onBack}>
            <Icon name="arrow-back" size={30} color="#2D9CDB" />
          </LeftButton>

        ) : (
          <BlankView color={blankViewColor}>
          </BlankView>
        )}
        <TitleTextView>
          <TitleText color={MainTextColor}>
            {MainText}
          </TitleText>

        </TitleTextView>
        {hasRightButton ? (
          <RightButton onPress={onPressRightButton} color={RightButtonBackgroundColor}>
            <Icon name="settings" size={30} color={RightButtonIconColor ? RightButtonIconColor : "#2D9CDB"}/>
          </RightButton>
        ) : (
          <BlankView color={blankViewColor}>
          </BlankView>
        )}
      </HeaderContainer>
      <BodyContainer>
       {children}
      </BodyContainer>
      <FooterContainer color={footerColor}>
        {hasButton ? 
        (<Button
            buttonColor={buttonColor}
            buttonText={buttonText}
            onPress={onPress}
          />
        ) : null}
      </FooterContainer>
    </Container>
  );
}

export default ScreenTemplate;
