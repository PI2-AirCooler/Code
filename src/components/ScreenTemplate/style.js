import styled from 'styled-components/native';
import normalize from 'react-native-normalize';

export const Container = styled.View`
  flex: 1;
  align-items: center;
  justify-content: center;
  /* background-color: #2D9CDB; */
`;

export const HeaderContainer = styled.View`
  flex: 0.15;
  align-items: center;
  width: 100%;
  justify-content: center;
  background-color: #F2F2F2;
`;

export const TitleText = styled.Text`
  color: #2D9CDB;
  font-size: ${normalize(35)}px;
  font-weight: bold;
  position: absolute;
`;

export const BodyContainer = styled.View`
  flex: 1;
  width: 100%;
  align-items: center;
  justify-content: center;
  background-color: #F2F2F2;
`;


export const AddDeviceButton = styled.TouchableOpacity`
  width: 70%;
  height: ${normalize(50)}px;
  justify-content: center;
  align-items: center;
  border-radius: ${normalize(7)}px;
  margin-bottom: ${normalize(-500)}px;
  background-color: #57B0E2
`;

export const AddDeviceButtonText = styled.Text`
  color: #FFF;
  font-size: ${normalize(20)}px;
  font-weight: bold;
`;
