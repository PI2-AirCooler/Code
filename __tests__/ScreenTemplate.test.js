/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import ScreenTemplate from '../src/components/ScreenTemplate';
 
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';
 
 it('ScreenTemplate renders correctly', () => {
   const tree = renderer
     .create(<ScreenTemplate />)
     .toJSON();
   expect(tree).toMatchSnapshot();
 });