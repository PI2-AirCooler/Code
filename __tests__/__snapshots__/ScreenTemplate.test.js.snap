// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenTemplate renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F2F2F2",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 0.15,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F2F2F2",
            "borderRadius": 14,
            "height": 100,
            "justifyContent": "center",
            "width": "15%",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#000",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "50%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2D9CDB",
              "fontSize": 70,
              "fontWeight": "bold",
              "position": "absolute",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F2F2F2",
            "borderRadius": 14,
            "height": 100,
            "justifyContent": "center",
            "width": "15%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "//": "center",
          "alignItems": "center",
          "backgroundColor": "#F2F2F2",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F2F2F2",
          "flexBasis": 0,
          "flexGrow": 0.15,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#57B0E2",
          "borderRadius": 14,
          "height": 100,
          "justifyContent": "center",
          "opacity": 1,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 40,
              "fontWeight": "bold",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
